### ISSUE TITLE
Custom notifications keep reverting to default, and switch themselves off

### ISSUE THREAD

**Issue by umbellipher** [signalapp/Signal-Android#11666]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Custom notifications keep reverting to default, and switch themselves off.

### Steps to reproduce
- open conversation settings
- set custom notifications to 'on'
- wait a time
- custom notifications switch themselves off again.

**Actual result:** Custom notifications switch themselves off again.
**Expected result:** Custom notifications should remain as user-set.

### Device info

**Device:** moto g30 (XT2129-2)
**Android version:** 11
**Signal version:** 5.24.14 (beta)

**Comment by fourmajor** [comment]
Also experiencing this:
Device: Pixel 6
Android version: 12
Signal version: 5.27.2

**Comment by geoffro10** [comment]
Same here.
Pixel 5
Android 12
Signal version: 5.27.13

**Comment by cody-signal** [comment]
Debuglogs would be greatly aid in investigating this issue, especially if taken right after noticing the change.

**Comment by geoffro10** [comment]
It just happened again so I grabbed the debug log. 

https://debuglogs.org/217d7146fddce3b6387928d70db68e2e9ddf109404c863123a6bd3e61dd8b37c

**Comment by cody-signal** [comment]
Okay, definitely something going on when restarting the app and it performs the channel consistency check. Are you doing only what OP reported or something else?

Mainly, need to know if you are making notification channel changes in the app settings for things like sound/vibrate/priority/bubble or only doing notification changes in the app via the settings screens.

**Comment by geoffro10** [comment]
I just tried to duplicate it and here's what I did.

Open the app
Open message for specific person
Open conversation settings
Open Sounds and notifications.... confirmed that custom notifications says "on"
Open custom notifications and confirmed the sound is "Fairydust sound"
Backed out to main screen and repeated from step two and now the notification sound says "default"
Immediately pulled a debug log.

https://debuglogs.org/94e06fcb0b9986730533b6d2c65154c979e669ba07645507d4a30f8c229ebb3e

Please let me know if there's anything else I can do to help track this down.

**Comment by cody-signal** [comment]
@geoffro10 Can you do the following:

Open the app
Goto convo settings for custom notifications
Enable custom notifications and set a sound
Open Android System settings for Signal and go to notifications
See if there are more than 1 channel for the chat you just set a sound for

**Comment by geoffro10** [comment]
I didn't see more than one for the one I just added. However, I see multiple entries for those contacts that I keep seeing it revert. I turned off custom notifications for one, exited out, came back and the custom notification was back on again. Looked in Android System and the duplicate for that person was gone. 

Another thing I noticed, when I added a custom notification for someone who didn't have it before, it created one in the Android System but the sound listed was still default. When I went back to Signal, it still showed custom notifications on and the specific sound still showed.

Looks to me that the duplicates are causing an issue.

**Comment by geoffro10** [comment]
Here's another update... I'm able to remove the duplicate entries by going back in to Signal and turning off custom notifications again. All seems to work when there's only one entry. I don't know how they got duplicated so I hope you do and can fix it.

**Comment by geoffro10** [comment]
Glad I could help. Let me know if you need any other help with it from my limited ability. LOL

**Comment by cody-signal** [comment]
There will be a hopeful fix in 5.29 to make it harder to create duplicate channels.

Did you happen to do anything with bubbles/priorities with some of these contacts that have been problematic?

**Comment by geoffro10** [comment]
Nope, the only thing I did was restore from a backup on to a new phone.

**Comment by major-mayer** [comment]
I have the problem, that when I disable vibration in the notification settings and open/close the window, it keeps re-enabling all the time.
Is this bug related or do I need to create a separate issue for that?

I am using 5.28.7

**Comment by major-mayer** [comment]
At least the vibration resetting thing seems to be still present in Signal 5.32.14

**Comment by youngestdj** [comment]
I am also facing this issue. If you switch off vibration in notification settings then hit back and open notification settings again you'll see it enabled again. Basically you can't switch off vibration in notifications. v5.34.10 Android 12

**Comment by CurtisTheGreat** [comment]
I'm running into this issue as well.

OnePlus 8T+5G, Signal 5.35.3, Android 11

**Comment by kouc52** [comment]
@cody-signal version 5.39.3 on a One+ 8T Pro

**Comment by cody-signal** [comment]
Thanks, @kouc52 can you submit a debuglog after noticing the switch.

**Comment by Kuladeep1996** [comment]
facing similar issue, custom notifications not working, only default works
Device: Realme 7
Android version: 11
Signal version: v5.42.7

**Comment by strugee** [comment]
@Kuladeep1996 please submit a debug log after noticing the switch, otherwise there's no way to diagnose what's going wrong

**Comment by major-mayer** [comment]
I am using 5.44.2 and wasn't able to reproduce neither the bug with the custom notifications nor the thing with the (global) vibration settings that i reported above.
Everything else should deserve a new issue I guess.

**Comment by cody-signal** [comment]
@fleapower was this for just a single contact? I see a log line indicating we couldn't find a channel configured in app in the system list of channels but it's only for a single contact, not the general channel used for messages

**Comment by fleapower** [comment]
I'm going to plead ignorance on that one...I'm not sure how to differentiate between a general channel used for messages and any other channels.  I noticed the change for one contact, but it happens to any/all contacts when I set custom notifications - they eventually revert back to non-customized notifications.

I only have three contacts with custom notifications.  I will periodically check throughout the day to see if they revert back to non-customized.  When I notice it, I will submit a new debug log - might capture some additional info.

**Comment by cody-signal** [comment]
Thanks, sorry for the jargon, but you got to what I was curious about. Knowing if your custom notification was configured for a single contact/conversation vs for all conversations. What I see in the logs suggests that we have a custom notification channel configured for a single conversation in app, but when we ask the OS if that channel still exists, it says no, so we disassociate it as we assume something was changed in the device's system settings for the app.

**Comment by fleapower** [comment]
Hmm...is there something I (or any other end user encountering this) could be doing to cause the app to mistakenly not find the channel and remove the custom notifications?

**Comment by fleapower** [comment]
I had a conversation revert back to non-custom notifications within the last ten minutes of this debug log bring created.  I don't know if it is connected, but I archived the conversation prior to the reversion.  When I received a new message in the conversation and it was automatically un-archived, it was the normal notification, not the custom notification.

https://debuglogs.org/android/5.50.4/88f747025661e9d5b14e81f9d62a26ecc97686571651cf0c0133890939715d02

**Comment by cody-signal** [comment]
This shouldn't matter, but may be an interesting data point, but are the contacts that change, are they SMS-only contacts or are they Signal?

**Comment by fleapower** [comment]
> This shouldn't matter, but may be an interesting data point, but are the contacts that change, are they SMS-only contacts or are they Signal?

They are Signal and SMS contacts.

**Comment by fleapower** [comment]
> I had a conversation revert back to non-custom notifications within the last ten minutes of this debug log bring created. I don't know if it is connected, but I archived the conversation prior to the reversion. When I received a new message in the conversation and it was automatically un-archived, it was the normal notification, not the custom notification.
> 
> https://debuglogs.org/android/5.50.4/88f747025661e9d5b14e81f9d62a26ecc97686571651cf0c0133890939715d02

Did the latest debug log show anything?

**Comment by fleapower** [comment]
Could really use this bug being fixed - just missed two time sensitive texts today because the notifications reverted back to the non-custom notification on the app.

**Comment by Crapshit** [comment]
I'm having a similar issue:
If I archive a conversation, it seems that the custom notification settings will be removed / disabled.
I only set some chats to "no sound / no vibration". After archiving and restoring these settings are no longer present.
Signal version 5.50.4, Pixel 6 Pro Android 13

**Comment by ElijahLynn** [comment]
I've been having this issue for months on a Pixel 7. And I finally was able to search the internet and find this issue. For me, it only happens to some contacts that I have custom notifications enabled for it within signal. And it won't revert for all the contacts that I have custom notifications for just one or maybe two of them. And then I just go in and turn it back on to custom notifications and it works fine for a while. I will try to look at debug logs and stuff the next time it happens.

**Comment by jayb-g** [comment]
Found [this](https://www.reddit.com/r/signal/comments/x31so5/sygnal_keeps_enabling_custom_notifications_for/?rdt=64715) and turns out be true that _MOST_ custom/notifications related issues are due to bubbling feature in Android. Turn it off and auto-resetting of custom notifications goes away.

**Comment by wertzuz** [comment]
#12353 #12126
BTW any reason Signal masseges are not priority messages?
